python/ray/data/tests/test_object_gc.py

```diff
@@ -235,6 +234,29 @@ def consume(p):
     assert "Spilled" not in meminfo, meminfo
 
 
+def test_global_bytes_spilled(shutdown_only):
+    # The object store is about 90MB.
+    ctx = ray.init(object_store_memory=90e6)
+    # The size of dataset is 500*(80*80*4)*8B, about 100MB.
+    ds = ray.data.range_tensor(500, shape=(80, 80, 4), parallelism=100).materialize()
+
+    with pytest.raises(AssertionError):
+        check_no_spill(ctx, ds.repeat())
```

Comment:
Out of scope for this PR, but `repeat` is deprecated and might get removed in an upcoming release. We should rewrite the tests in this module to not use `DatasetPipeline`.